From 8a8c7a1aa514eb3c279e77368352076f06bf50c0 Mon Sep 17 00:00:00 2001
From: Iain Lane <iain@orangesquash.org.uk>
Date: Mon, 11 May 2015 14:19:52 +0100
Subject: [PATCH] glocalfilemonitor: Send DELETED event when there is no
 'other' file
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The documentation states "If using […] G_FILE_MONITOR_SEND_MOVED flag
and event_type is G_FILE_MONITOR_EVENT_MOVED, […] other_file will be set
to a GFile containing the new path. In all the other cases, other_file
will be set to NULL."

There's code in the wild which expects a non-null other_file when it
gets a MOVED event.

When moving a file outside of a monitored directory, send DELETED. This
is the pre-2.45 behaviour.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=749225
---
 gio/glocalfilemonitor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/glocalfilemonitor.c b/gio/glocalfilemonitor.c
index 8a5c442..31ba1a7 100644
--- a/gio/glocalfilemonitor.c
+++ b/gio/glocalfilemonitor.c
@@ -385,7 +385,7 @@ g_file_monitor_source_handle_event (GFileMonitorSource *fms,
       g_assert (!rename_to);
       if (fms->flags & G_FILE_MONITOR_WATCH_MOVES)
         g_file_monitor_source_send_event (fms, G_FILE_MONITOR_EVENT_MOVED_OUT, child, other);
-      else if (fms->flags & G_FILE_MONITOR_SEND_MOVED)
+      else if (fms->flags & G_FILE_MONITOR_SEND_MOVED && other)
         g_file_monitor_source_send_event (fms, G_FILE_MONITOR_EVENT_MOVED, child, other);
       else
         g_file_monitor_source_send_event (fms, G_FILE_MONITOR_EVENT_DELETED, child, NULL);
-- 
2.1.4

